editor/editor_help_search.cpp

```diff
@@ -161,7 +161,9 @@ void EditorHelpSearch::popup_dialog() {
 void EditorHelpSearch::popup_dialog(const String &p_term) {
 	// Restore valid window bounds or pop up at default size.
 	Rect2 saved_size = EditorSettings::get_singleton()->get_project_metadata("dialog_bounds", "search_help", Rect2());
-	if (saved_size != Rect2()) {
+	Rect2 screen_size = DisplayServer::get_singleton()->screen_get_usable_rect(get_window()->get_current_screen());
+
+	if (saved_size != Rect2() || !screen_size.encloses(saved_size)) {
```

Comment:
The second condition is only considered if `saved_size == Rect2()`, so this check should always fail, no?

Reply:
Yeah 🤦‍

Reply:
Boolean combinations are damn tricky, do this myself all the time